packages/platform-android/native_modules.gradle

```diff
@@ -187,6 +247,85 @@ class ReactNativeModules {
     }
   }
 
+  void generateAndroidMkFile(File outputDir, String generatedFileName, String generatedFileContentsTemplate) {
+    ArrayList<HashMap<String, String>> packages = this.reactNativeModules
+    String packageName = this.packageName
+    String libraryIncludes = ""
+    String libraryModules = ""
+
+    if (packages.size() > 0) {
+      libraryIncludes = packages.collect {
+        "include ${it.androidMkPath}"
+      }.join('\n')
+      libraryModules = packages.collect {
+        "libreact_codegen_${it.libraryName}"
+      }.join(" ")
+    }
+
+    String generatedFileContents = generatedFileContentsTemplate
+      .replace("{{ libraryIncludes }}", libraryIncludes)
+      .replace("{{ libraryModules }}", libraryModules)
+
+    outputDir.mkdirs()
+    final FileTreeBuilder treeBuilder = new FileTreeBuilder(outputDir)
+    treeBuilder.file(generatedFileName).newWriter().withWriter { w ->
+      w << generatedFileContents
+    }
+  }
+
+  void generateRncliCpp(File outputDir, String generatedFileName, String generatedFileContentsTemplate) {
+    ArrayList<HashMap<String, String>> packages = this.reactNativeModules
+    String rncliCppIncludes = ""
+    String rncliCppModuleProviders = ""
+    String rncliCppComponentDescriptors = ""
+
+    if (packages.size() > 0) {
+      rncliCppIncludes = packages.collect {
+        def result = "#include <${it.libraryName}.h>"
+        if (it.componentDescriptors.size() > 0) {
+          result += "\n#include <react/renderer/components/${it.libraryName}/ComponentDescriptors.h>"
```

Comment:
Two things here:
1. Please export to a constant the `react/renderer/components/` and `ComponentDescriptors.h`
2. There might be some escaping to do. I'll check and get back to you